mgrachev/reverse linked list/Program.cs

```diff
@@ -0,0 +1,92 @@
+﻿/* Reverse Linked List
+
+Given the head of a singly linked list, reverse the list, and return the reversed list.
+
+Example:
+Input: head = [1,2,3,4,5]
+Output: [5,4,3,2,1]
+
+
+-------------------------------
+
+
+Solution:
+
+Here is a layout for our linked list:
+(1) -> (2) -> (3) -> (4) -> (5) ->
+
+
+We will iterate through our linked list with a while cycle.
+
+We will need two pointers to reverse our linked list.
+One for  "previously" visited node (prev), and one for "current" node (cur). 
+
+We will start with prev = null and cur = first node.
+
+     (1) -> (2) -> (3) -> (4) -> (5) ->
+^     ^
+|     |
+prev  cur
+
+On each iteration step we will retarget "current" node onto the "previous" node, 
```

Comment:
Such an elegant algorithm :) and thanks as usual for detailed descriptions, really helps for noobs like me to immediately understand what's happening.